dev/dvm/lib/dvm/project/erlang.rb

```diff
@@ -165,42 +166,70 @@ def do_build
       run_command("make devrel -j #{concurrency}",    "Building... ", cwd: project_dir, env: { "USE_SYSTEM_GECODE" => "1"})
     end
 
-    def enable_cover(modules = nil)
-      say "Disabling sync to prevent conflicts in loaded beam files."
-      update("pause")
-        run_command("#{erl_command} -eval \"rpc:call('#{node}', sync, go, []).\" -s erlang halt", "Resuming sync on #{node}")
-        run_command("#{erl_command} -eval \"rpc:call('#{node}', sync, go, []).\" -s erlang halt", "Resuming sync on #{node}")
-      say "Cover-compiling app beams."
-      # Run these guys next - need to do it by hand via eval!
-      eval_cmd = <<-EOM
-F = fun() ->
-  Dirs = file:wildcard("/host/#{project_name}/apps/*/ebin\") ++ ["/host/#{project_name}/ebin"]
-  [cover:compile_beam_dir(Dir) || Dir <- Dirs]
-end,
-spawn(#{node}, F).
-EOM
-      run_command("#{erl_command} -eval \"#{eval_cmd}\" -s erlang halt", "Cover-compiling modules on the erlang node.")
-    end
-    def dump_cover(modules = nil)
-      # TODO
+    def cover(action, modulename, options)
```

Comment:
I like that, thanks. I found myself missing pattern matching in fun heads .... :D